syweb/webclient/app/app.js

```diff
@@ -125,7 +129,7 @@ matrixWebClient.run(['$location', '$rootScope', 'matrixService', function($locat
     // If user auth details are not in cache, go to the login page
     $rootScope.$on("$routeChangeStart", function(event, next, current) {
         if (!matrixService.isUserLoggedIn() && $location.path() !== "/login" &&
-                $location.path() !== "/register") {
+                $location.path() !== "/register" && $location.path() !== "/reset-password") {
```

Comment:
It would probably be cleaner at this point to do:
```javascript
var unauthenticatedPages = ["/register", "/reset-password", "/login"];
if (!matrixService.isUserLoggedIn() && unauthenticatedPages.indexOf($location.path()) === -1) {
  ...
}
```